source/common/json/json_streamer.h

```diff
@@ -373,11 +385,14 @@ class Streamer {
   void addBool(bool b) { response_.add(b ? Constants::True : Constants::False); }
 
   /**
-   * Adds a constant string to the output stream. The string must outlive the
-   * Streamer object, and is intended for literal strings such as punctuation.
+   * Adds a pre-sanitized string or which doesn't require sanitizing to the output stream.
+   * NOTE: use this with care as it bypasses the sanitization process and may result in
+   * invalid JSON. If you are not sure if the string is already sanitized, use addString()
+   * or addSanitized() instead.
    */
-  void addConstantString(absl::string_view str) { response_.add(str); }
+  void addWithoutSanitizing(absl::string_view str) { response_.add(str); }
```

Comment:
do we have an easy utility function such that we could ASSERT it's sanitized?

Reply:
should be easy: run the sanitizer and check it's the same.

Though it might be simpler to just assert that `needs_slow_sanitizer_[c]` is false for every char in str -- that'd have to be exported under `ifndef NDEBUG` from json_sanitizer.cc.

Reply:
> do we have an easy utility function such that we could ASSERT it's sanitized?

I think may we I won't use this method (depends on the final result of #35545)

So, I inclined to leave it as this. If finally this is not used, I can create a new PR to mark this as private method and only used to push delimiters to output buffer.